src/h5web/visualizations/heatmap/hooks.ts

```diff
@@ -1,10 +1,53 @@
-import React, { useCallback, useRef, useEffect } from 'react';
+import React, { Ref, useCallback, useRef, useEffect } from 'react';
 import { Vector3 } from 'three';
 import { ReactThreeFiber, PointerEvent, useThree } from 'react-three-fiber';
 import { clamp } from 'lodash-es';
+import { useMeasure } from 'react-use';
+import { useHeatmapState } from './store';
 
 const ZOOM_FACTOR = 0.95;
 
+export function useHeatmapSize<T>(
+  dims: [number, number],
+  axisOffset: [number, number]
+): [(elem: T) => void, { width: number; height: number } | undefined] {
+  const { hasOrthoCoords } = useHeatmapState();
+  const [wrapperRef, { width, height }] = useMeasure();
+
+  if (width === 0 && height === 0) {
+    return [wrapperRef, undefined];
+  }
+
+  if (!hasOrthoCoords) {
+    return [wrapperRef, { width, height }];
+  }
+
+  const [rows, cols] = dims;
+  const dimsRatio = rows / cols;
+
+  const [leftAxisWidth, bottomAxisHeight] = axisOffset;
+  const actualWidth = width - leftAxisWidth;
+  const actualHeight = height - bottomAxisHeight;
+
+  const shouldAdjustWidth = actualWidth >= actualHeight * dimsRatio;
```

Comment:
I would add a comment here explaining the need for this boolean

Reply:
Phew, I struggled with this one. Hope it is clear now.